Client/src/App.jsx

```diff
@@ -39,6 +39,17 @@ import { useDispatch } from "react-redux";
 import { getAppSettings, updateAppSettings } from "./Features/Settings/settingsSlice";
 import { logger } from "./Utils/Logger"; // Import the logger
 import { networkService } from "./main";
+import { setMode } from "./Features/UI/uiSlice";
+
+const getPreferredTheme = () => {
+	try {
+		return window?.matchMedia?.('(prefers-color-scheme: dark)')?.matches? 'dark': 'light';
+	}
+	catch {
+		return 'light';
+	}
+}
```

Comment:
Thank you for your time working on this issue, @aarya-16 !

This is a good path, but doing this in the app directory can bring some performance issues. I suggest you do the check using a ternary inside of the UI slice. so instead of defining the inital theme as light, we will be getting the inital one dynamically at the store level. With that we won't need to call a state update with the use effect.

Reply:
I'll implement those changes, @marcelluscaio. I do have one question though:
Do you want me to add the event listeners to handle dynamic user system theme changes in the UI Slice as well ?

Reply:
Good point! I don't think we need the event listener. All we want is to assess user's initial preference. The rest is being dealt with through the settings menu. Do you see a scenario in which that would be necessary?

Reply:
The only scenario I see is if someone changes their browser or system theme settings while the website is active. But even in that case, a page reload would do the trick. So I guess I'll just make the changes without the event listener.

Reply:
Great! I agree with you, I don't think users expect a live reload according to system settings, and we would get loose of an event listener.